bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/transport/PanamaxFurmanSocket.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.panamaxfurman.internal.transport;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanCommunicationEventListener;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanConnectionBrokenEvent;
+import org.openhab.binding.panamaxfurman.internal.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wraps the communication to a Panamax/Furman Power Conditioner with a BlueBOLT-CV1 or BlueBOLT-CV2 card via telnet
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public class PanamaxFurmanSocket {
+    private static final int CONNECT_TIMEOUT_MILLIS = (int) TimeUnit.SECONDS.toMillis(10);
+    private static final int INFINITE_TIMEOUT = 0;
+
+    protected final Object ioMutex = new Object();
+    private final Logger logger = LoggerFactory.getLogger(PanamaxFurmanSocket.class);
+
+    private final InputStreamReaderThread inputStreamReaderThread;
+    private final BufferedWriter writer;
+    private final PanamaxFurmanCommunicationEventListener eventListener;
+
+    private final String connectionName;
+
+    private Socket telnetSocket;
+
+    /**
+     * Creates a new connection to the power conditioner device
+     */
+    public PanamaxFurmanSocket(String connectionName, String address, int port,
+            PanamaxFurmanCommunicationEventListener eventListener) throws IOException, InterruptedException {
+        this.eventListener = eventListener;
+        this.connectionName = connectionName;
+        synchronized (ioMutex) {
+            logger.trace("Attempting to connect via telnet to {}:{}  @{}", address, port, connectionName);
+            try {
+                telnetSocket = new Socket();
+                telnetSocket.connect(new InetSocketAddress(address, port), CONNECT_TIMEOUT_MILLIS);
+                telnetSocket.setKeepAlive(true);
+                // Block forever on reads
+                telnetSocket.setSoTimeout(INFINITE_TIMEOUT);
```

Comment:
It think it is better to set this to something high rather then 'infinite'

Reply:
why?  I don't see a point in arbitrarily timing out the connection for no reason

Reply:
In general it is a good practive to timeout connections. Preventing hanging connections to exists/block forever. I'm not 100% sure if that makes sense in this context, but i prefer to have some timeout so that connectivity issues do not get hidden. I leave it to you to decide.

Reply:
so the more I thought about it, I agree that the timeout should not be inifinite.  From what I can tell, the device sends updates relatively frequently, so if we're not getting those, it's probably a good idea to reset the connection.  Let me do some analysis on that to determine a good default timeout.  Will probably make this configurable as an advanced setting as well